openssl/src/ssl/connector.rs

```diff
@@ -552,12 +552,19 @@ cfg_if! {
                 let hostname_label = &hostname[..hostname_label_end];
 
                 // check the prefix of the first label
-                if !hostname_label.starts_with(wildcard_prefix) {
+                if !hostname_label[..wildcard_location].eq_ignore_ascii_case(wildcard_prefix) {
```

Comment:
Do we know at this point that hostname_label is at least as long as wildcard_location?

Reply:
indeed, we do not. Good catch. There's [a check for that case](https://github.com/sfackler/rust-openssl/pull/1378/commits/76d996c380334217efba23ceddc050c045ababbb#diff-1b6ced8dfb91d53c0d3c93878fca256ae1431551161033aa3a264992e018829bR555-R559) and [more tests](https://github.com/sfackler/rust-openssl/pull/1378/commits/76d996c380334217efba23ceddc050c045ababbb#diff-1b6ced8dfb91d53c0d3c93878fca256ae1431551161033aa3a264992e018829bR601-R605) for partial wildcards now.